specs/agents/tracing-api-otel.md

```diff
@@ -0,0 +1,54 @@
+## OpenTelemetry API (Tracing)
+
+[OpenTelemetry](https://opentelemetry.io) (OTel in short) provides a vendor-neutral API that allows to capture tracing, logs and metrics data.
+
+Agents may provide a bridge implementation of OpenTracing Tracing API following this specification.
+
+Bridging here means that for each OTel span created with the API, a native span/transaction will be created and sent to APM server.
+
+From the perspective of the application code calling the OTel API, the delegation to a native span/transaction is transparent.
+Also, this means that any OTel processors will be bypassed and ignored by the bridge.
+
+### Attributes mapping
+
+OTel relies on key-value pairs for span attributes. Keys and values are protocol-specific and are defined in [semantic convention](https://github.com/open-telemetry/opentelemetry-specification/tree/main/specification/trace/semantic_conventions) specification.
+
+In order to minimize the mapping complexity in agents, most of the mapping between OTel attributes and agent protocol will be delegated to APM server:
+- All OTel span attributes should be captured as-is and written to agent protocol.
+- APM server will handle the mapping between OTel attributes and their native transaction/spans equivalents
+- Some native span/transaction attributes will still require mapping within agents for [compatibility with existing features](#compatibility-mapping)
+
+OpenTelemetry attributes should be stored in `otel.attributes` as a flat key-value pair mapping added to `span` and `transaction` objects:
+```json
+{
+  // [...] other span/transaction attributes
+  "otel": {
+    "attributes": {
+      "db.system": "mysql",
+      "db.statement": "SELECT * from table_1"
+    }
+  }
+}
+```
+
+When the APM server version does not support `otel.attributes` property, agents should use `labels` as fallback with OTel attribute
+name as key.
```

Comment:
When sending events via this bridge, will all the required fields still be populated by the apm agents or do we need to change the json spec beyond adding the `otel.attributes`?

Reply:
These are the only required properties, right?
https://github.com/elastic/apm-server/blob/889da019ef4fd88540e0b6a8443143727acab111/docs/spec/v2/transaction.json#L1008-L1014
https://github.com/elastic/apm-server/blob/889da019ef4fd88540e0b6a8443143727acab111/docs/spec/v2/span.json#L762-L769

These should all be populated. The only thing agents need be aware of is to infer `type`.
Hypothetically speaking, could agents sent an empty string for the `type` and rely on APM Server to do the inference based on the `otel.attributes`?

Reply:
+1 on keeping the current required fields, it is fine to use empty strings.

Agents will have to infer `type` anyway, but maybe as a general rule we can add that any attribute set into `otel.attributes` should override its IntakeV2 counterpart when there is a direct mapping.
For example, if we have `span.otel.attribute['span.type']` provided, it will override `span.type` even if the latter is provided, hence having an empty `span.type` would be fine.

Thus, for agents, that could allow to avoid duplicating some attributes (`name`, `type`, ...) when those have a direct mapping.

Reply:
I have found the mapping rules that APM Server uses to infer type/subtype/action.

* Transactions: https://github.com/elastic/apm-server/blob/a275ed9dc843762b42737a108b47533afa1eef70/processor/otel/traces.go#L404-L414
* Spans: https://github.com/elastic/apm-server/blob/a275ed9dc843762b42737a108b47533afa1eef70/processor/otel/traces.go#L715-L756

Reply:
Mapping has been implemented in APM server 7.16, so we can resolve this conversation thread for now.